fluss-client/src/main/java/org/apache/fluss/client/metadata/MetadataUpdater.java

```diff
@@ -316,14 +335,61 @@ private static Cluster initializeCluster(Configuration conf, RpcClient rpcClient
         return cluster;
     }
 
-    private static Cluster tryToInitializeCluster(RpcClient rpcClient, InetSocketAddress address)
+    @VisibleForTesting
+    static @Nullable Cluster tryToInitializeClusterWithRetries(
+            RpcClient rpcClient,
+            ServerNode serverNode,
+            AdminReadOnlyGateway gateway,
+            int maxRetryTimes)
+            throws Exception {
+        int retryCount = 0;
+        while (retryCount <= maxRetryTimes) {
+            try {
+                return tryToInitializeCluster(gateway);
+            } catch (Exception e) {
+                Throwable cause = stripExecutionException(e);
+                if (!(cause instanceof StaleMetadataException
+                        || cause instanceof NetworkException)) {
+                    throw e;
+                }
+
+                if (retryCount >= maxRetryTimes) {
```

Comment:
Add comment above `// in case of bootstrap is recovering, we should retry to connect`